File: cabinet_plugin/android/src/main/kotlin/com/flutter/cabinet_plugin/entities/CellData.kt
package com.flutter.cabinet_plugin.entities

/**
 * 格子的状态
 * @param state 锁状态 0开启 1关闭
 * @param  pressure 气压
 * @param  temperature 温度
 * @param humidity 湿度
 */
data class CellData(
    val state: Int,
    val pressure: Double,
    val temperature: Double,
    val humidity: Double
)

File: cabinet_plugin/lib/CellData.dart
class CellData {
  //门锁状态  0开启 1关闭
  int state;

  //气压
  double pressure;

  //温度
  double temperature;

  //湿度
  double humidity;

  CellData({this.state, this.pressure, this.temperature, this.humidity});

  factory CellData.fromMap(Map<String, dynamic> json) {
    return CellData(
      state: json['state'] as int,
      pressure: json['pressure'] as double,
      temperature: json['temperature'] as double,
      humidity: json['humidity'] as double,
    );
  }
}
